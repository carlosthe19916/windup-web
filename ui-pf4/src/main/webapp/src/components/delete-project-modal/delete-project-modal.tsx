import React from "react";
import { useState, useEffect } from "react";
import {
  Button,
  Modal,
  ButtonVariant,
  ModalVariant,
  TextInput,
} from "@patternfly/react-core";

export interface DeleteProjectModalProps {
  isModalOpen: boolean;

  projectTitle: string;
  matchText: string;

  inProgress?: boolean;
  onDelete: () => void;
  onCancel: () => void;
}

export const DeleteProjectModal: React.FC<DeleteProjectModalProps> = ({
  isModalOpen,
  projectTitle,
  matchText,
  inProgress,
  onDelete,
  onCancel,
}) => {
  const [inputMatchValue, setInputMatchValue] = useState("");
  const [allowDeletion, setAllowDeletion] = useState(false);

  useEffect(() => {
    setAllowDeletion(matchText.toLowerCase() === inputMatchValue.toLowerCase());
  }, [inputMatchValue, matchText]);

  const handleDelete = () => {
    onDelete();
  };

  const handleCancel = () => {
    onCancel();
  };

  return (
    <Modal
      variant={ModalVariant.small}
      title={`Confirm project deletion`}
      isOpen={isModalOpen}
      onClose={handleCancel}
      actions={[
        <Button
          key="delete"
          variant={ButtonVariant.danger}
          isDisabled={!allowDeletion || inProgress}
          onClick={handleDelete}
        >
          Delete
        </Button>,
        <Button
          key="cancel"
          variant="link"
          onClick={handleCancel}
          isDisabled={inProgress}
        >
          Cancel
        </Button>,
      ]}
    >
      <div className="pf-c-content">
        <p>
          Are you sure you want to delete the project '{projectTitle}'? This
          will delete all resources associated with '{projectTitle}' and cannot
          be undone. Make sure this is something you really want to do!
        </p>

        <p>Type '{matchText}' to confirm.</p>

        <TextInput
          value={inputMatchValue}
          type="text"
          onChange={(value) => setInputMatchValue(value)}
          aria-label="Text input match"
        />
      </div>
    </Modal>
  );
};
