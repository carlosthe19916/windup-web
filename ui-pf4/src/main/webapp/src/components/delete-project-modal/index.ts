export { DeleteProjectModal } from "./delete-project-modal";
