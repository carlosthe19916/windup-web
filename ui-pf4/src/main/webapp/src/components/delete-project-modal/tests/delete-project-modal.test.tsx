import React from "react";
import { shallow } from "enzyme";
import { DeleteProjectModal } from "../delete-project-modal";

describe("DeleteProjectModal", () => {
  it("Renders without crashing", () => {
    const wrapper = shallow(
      <DeleteProjectModal
        projectTitle="master"
        matchText="delete"
        isModalOpen={true}
        onDelete={jest.fn}
        onCancel={jest.fn}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
