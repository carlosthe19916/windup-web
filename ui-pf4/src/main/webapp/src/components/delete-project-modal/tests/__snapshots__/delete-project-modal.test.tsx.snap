// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteProjectModal Renders without crashing 1`] = `
<Modal
  actions={
    Array [
      <Button
        isDisabled={true}
        onClick={[Function]}
        variant="danger"
      >
        Delete
      </Button>,
      <Button
        onClick={[Function]}
        variant="link"
      >
        Cancel
      </Button>,
    ]
  }
  appendTo={[Function]}
  aria-describedby=""
  aria-label=""
  aria-labelledby=""
  className=""
  hasNoBodyWrapper={false}
  isOpen={true}
  onClose={[Function]}
  ouiaSafe={true}
  showClose={true}
  title="Confirm project deletion"
  variant="small"
>
  <div
    className="pf-c-content"
  >
    <p>
      Are you sure you want to delete the project '
      master
      '? This will delete all resources associated with '
      master
      ' and cannot be undone. Make sure this is something you really want to do!
    </p>
    <p>
      Type '
      delete
      ' to confirm.
    </p>
    <ForwardRef
      aria-label="Text input match"
      onChange={[Function]}
      type="text"
      value=""
    />
  </div>
</Modal>
`;
